core/src/cg/group4/game_logic/stroll/Stroll.java

```diff
@@ -1,198 +1,269 @@
+     */
+    protected final void generatePossibleEvent() {
+        Random rnd = new Random();
+        if (rnd.nextDouble() < cEventThreshold) {
+            cEventGoing = true;
+            int chosenEvent = rnd.nextInt(2);
+            switch(chosenEvent) {
+                case(0):
+                    cEvent = new FishingStrollEvent();
+                    break;
+                case(1):
+                    cEvent = new TestStrollEvent();
+                    break;
+                default:
+                    cEvent = new TestStrollEvent();
+            }
+            cNewEventSubject.update(cEvent);
+        }
+    }
+
+    /**
+     * Handles completion of an event.
+     *
+     * @param rewards Reward(s) given on completion of an event
+     */
+    public final void eventFinished(final int rewards) {
+        Gdx.app.log(TAG, "Event completed! Collected " + rewards + " rewards.");
+
+        cEndEventSubject.update(rewards);
+
+        cRewards += rewards;
+        cEvent = null;
+        cEventGoing = false;
+
+        if (cFinished) {
+            Gdx.app.log(TAG, "Event finished and time is up, ending stroll.");
+            done();
+        }
+    }
+
+    /**
+     * Method that returns the amplifier for the event chance.
+     * @param state Movement state gotten from the AccelLib library.
+     * @return Integer used to amplify the chance of getting the event.
+     */
+    protected int getAmplifier(final AccelerationState state) {
+        for (Amplifier a : Amplifier.values()) {
+            if (a.cState == state) {
+                return a.cAmplifier;
+            }
+        }
+        return 0;
+    }
+
+
+    /**
+     * Method that gets called when the stroll has ended/completed.
+     */
+    public final void done() {
+        Gdx.app.log(TAG, "Stroll has ended. Collected " + cRewards + " rewards.");
+
+
+        StandUp.getInstance().getUpdateSubject().deleteObserver(this);
+
+        cEndStrollSubject.update(cRewards);
+        cEndStrollSubject.deleteObservers();
+
+        cStrollTimer.getStopSubject().deleteObserver(cStrollStopObserver);
+
+        StandUp.getInstance().endStroll();
+    }
+
+    /**
+     * Getter for the subject to subscribe to to get updated for the end of the stroll.
+     *
+     * @return Subject to subscribe to.
+     */
+    public final Subject getEndStrollSubject() {
+        return cEndStrollSubject;
+    }
+
+    /**
+     * Getter for the subject to subscribe to get updated for the start of a new event.
+     *
+     * @return Subject to subscribe to.
+     */
+    public final Subject getNewEventSubject(){
+        return cNewEventSubject;
+    }
+
+    /**
+     * Getter for the subject to subscribe to to get updated for the end of the event.
+     * @return Subject to subscribe to.
+     */
+    public final Subject getEndEventSubject() {
+        return cEndEventSubject;
+    }
+
+    /**
+     * Amplifier enum.
+     */
+    public enum Amplifier {

```

Comment:
Don't think this is the best solution but can not think of a better 123... This is a point to improve the code